pkg/fs/xfs.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs
+
+import (
+	"os/exec"
+	"strings"
+
+	"github.com/docker/docker/pkg/mount"
+
+	"github.com/vmware/vic/pkg/trace"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// XFS satisfies the Filesystem interface
+type XFS struct{}
+
+// Create an XFS filesystem manager
+func NewXFS() *XFS {
+	return &XFS{}
+}
+
+// Mkfs creates an xfs fs on the given device and applices the given label
+func (e *XFS) Mkfs(devPath, label string) error {
+	defer trace.End(trace.Begin(devPath))
+
+	log.Infof("Creating xfs filesystem on device %s", devPath)
+
+	// #nosec: Subprocess launching with variable
+	cmd := exec.Command("/sbin/mkfs.xfs", "-n", "ftype=1", "-L", label, devPath)
+
+	if output, err := cmd.CombinedOutput(); err != nil {
+		log.Errorf("vmdk storage driver failed to format disk %s: %s", devPath, err)
+		log.Errorf("mkfs output: %s", string(output))
+		return err
+	}
+	log.Debugf("Filesystem created on device %s", devPath)
+
+	return nil
+}
+
+// Mount mounts an xfs formatted device at the given path.  From the Docker
+// mount pkg, args must in the form arg=val.
+func (e *XFS) Mount(devPath, targetPath string, options []string) error {
```

Comment:
I'm curious to know if we couldn't just leave `Mount`/`Unmount`/`SetLabel` in `disk.go` and pull them out of `pkg/fs/ext4.go` and `pkg/fs/xfs.go` so there's a little less DRYness. None of them seem to require the receiver so I'm wondering if they can't just be consolidated.